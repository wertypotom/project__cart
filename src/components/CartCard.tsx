import React from 'react';
import { Button, Card, Container } from 'react-bootstrap';
import { TItemToggle, TStoreItems } from '../types/TStoreItems';
import { formatCurrency } from '../utilities/currencyFormater';

type Props = {
  item: TStoreItems;
  toggleItem: (id: number, type: TItemToggle) => void;
};

const CartCard = ({ item, toggleItem }: Props) => {
  const totalItems = item.amount * item.price;

  return (
    <Card className='d-flex'>
      <Card.Img
        src={item.imgUrl}
        height='100px'
        width='100'
        style={{ objectFit: 'cover' }}
      />
      <Card.Body style={{ gap: 5 }} className='d-flex align-items-center'>
        <div className='d-flex flex-column me-auto'>
          <span>{item.name}</span>
          <span className='text-muted'>{formatCurrency(item.price)}</span>
        </div>
        <span>{formatCurrency(totalItems)}</span>
        <Button
          onClick={() => toggleItem(item.id, 'reset')}
          style={{ height: 40 }}
          variant='outline-danger'
          className='rounded-2 d-flex align-align-items-center justify-content-center '
        >
          x
        </Button>
      </Card.Body>
    </Card>
  );
};

export default CartCard;
