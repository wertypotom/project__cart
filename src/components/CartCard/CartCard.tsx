import React, { useContext } from 'react';
import { Button, Card } from 'react-bootstrap';
import cartContext from '../../context/cartContext';
import { TStoreItems } from '../../types/TStoreItems';
import formatCurrency from '../../utilities/currencyFormater';

type Props = {
  item: TStoreItems;
};

function CartCard({ item }: Props) {
  const { toggleItem } = useContext(cartContext);
  const totalItems = item.amount * item.price;

  return (
    <Card className="d-flex">
      <Card.Img
        src={item.imgUrl}
        height="100px"
        width="100"
        style={{ objectFit: 'cover' }}
      />
      <Card.Body style={{ gap: 5 }} className="d-flex align-items-center">
        <div className="d-flex flex-column me-auto">
          <span>{item.name}</span>
          <span className="text-muted">{formatCurrency(item.price)}</span>
        </div>
        <span>{formatCurrency(totalItems)}</span>
        <Button
          onClick={() => toggleItem(item.id, 'reset')}
          style={{ height: 40 }}
          variant="outline-danger"
          className="rounded-2 d-flex align-align-items-center justify-content-center "
        >
          x
        </Button>
      </Card.Body>
    </Card>
  );
}

export default CartCard;
