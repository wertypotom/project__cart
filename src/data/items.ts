import { TStoreItems } from '../types/TStoreItems';

export const storeItems: TStoreItems[] = [
  {
    id: 1,
    name: 'Book',
    price: 10.99,
    imgUrl: '/imgs/book.jpg',
    amount: 0,
  },
  {
    id: 2,
    name: 'Computer',
    price: 1199,
    imgUrl: '/imgs/computer.jpg',
    amount: 0,
  },
  {
    id: 3,
    name: 'Banana',
    price: 1.05,
    imgUrl: '/imgs/banana.jpg',
    amount: 0,
  },
  {
    id: 4,
    name: 'Car',
    price: 14000,
    imgUrl: '/imgs/car.jpg',
    amount: 0,
  },
];
