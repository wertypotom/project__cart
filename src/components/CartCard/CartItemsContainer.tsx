import React, { useContext } from 'react';
import { Container } from 'react-bootstrap';
import cartContext from '../../context/cartContext';
import formatCurrency from '../../utilities/currencyFormater';
import CartCard from './CartCard';

function CartItemsContainer() {
  const { items } = useContext(cartContext);
  const totalPrice = items.reduce((r, i) => (r += i.amount * i.price), 0);

  return (
    <Container className="d-flex flex-column gap-2">
      {items.map((item) => {
        if (item.amount) {
          return <CartCard key={item.id} item={item} />;
        }

        return null;
      })}
      <div className="align-self-end fw-bold">
        Total: {formatCurrency(totalPrice)}
      </div>
    </Container>
  );
}

export default CartItemsContainer;
