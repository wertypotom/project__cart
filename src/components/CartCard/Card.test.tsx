import React from 'react';
import { describe, it } from 'vitest';
import CartItemsContainer from './CartItemsContainer';
import CartCard from './CartCard';
import { render } from '@testing-library/react';
import items from '../../data/items';

describe('Testing CartItemsContainer', () => {
  it('Renders CartItemsContainer conponent', () => {
    const component = render(<CartItemsContainer />);

    expect(component).toMatchSnapshot();
  });
});

describe('Testing Single Cart', () => {
  it('Renders CartItemsContainer conponent', () => {
    const component = render(<CartCard item={items[0]} />);

    expect(component).toMatchSnapshot();
  });
});
