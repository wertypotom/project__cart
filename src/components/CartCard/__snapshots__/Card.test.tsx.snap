// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Cart of Cards > Renders Cart conponent 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column gap-2 container"
      >
        <div
          class="align-self-end fw-bold"
        >
          Total: 
          0,00 $
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column gap-2 container"
    >
      <div
        class="align-self-end fw-bold"
      >
        Total: 
        0,00 $
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing CartItemsContainer > Renders CartItemsContainer conponent 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column gap-2 container"
      >
        <div
          class="align-self-end fw-bold"
        >
          Total: 
          0,00 $
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column gap-2 container"
    >
      <div
        class="align-self-end fw-bold"
      >
        Total: 
        0,00 $
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Single Cart > Renders CartItemsContainer conponent 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex card"
      >
        <img
          class="card-img"
          height="100px"
          src="/imgs/book.jpg"
          style="object-fit: cover;"
          width="100"
        />
        <div
          class="d-flex align-items-center card-body"
          style="gap: 5px;"
        >
          <div
            class="d-flex flex-column me-auto"
          >
            <span>
              Book
            </span>
            <span
              class="text-muted"
            >
              10,99 $
            </span>
          </div>
          <span>
            0,00 $
          </span>
          <button
            class="rounded-2 d-flex align-align-items-center justify-content-center  btn btn-outline-danger"
            style="height: 40px;"
            type="button"
          >
            x
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex card"
    >
      <img
        class="card-img"
        height="100px"
        src="/imgs/book.jpg"
        style="object-fit: cover;"
        width="100"
      />
      <div
        class="d-flex align-items-center card-body"
        style="gap: 5px;"
      >
        <div
          class="d-flex flex-column me-auto"
        >
          <span>
            Book
          </span>
          <span
            class="text-muted"
          >
            10,99 $
          </span>
        </div>
        <span>
          0,00 $
        </span>
        <button
          class="rounded-2 d-flex align-align-items-center justify-content-center  btn btn-outline-danger"
          style="height: 40px;"
          type="button"
        >
          x
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
